src/core/tactics.ml

```diff
@@ -32,40 +59,72 @@ let handle_tactic : Sig_state.t -> Proof.t -> p_tactic -> Proof.t =
   | _                   ->
   (* The other tactics may change the goals. *)
   (* Get the focused goal and the other goals. *)
-  let (g, gs) =
-    match ps.proof_goals with
-    | []    -> fatal tac.pos "There is nothing left to prove.";
-    | g::gs -> (g, gs)
+    if ps.proof_goals = [] then
+      fatal tac.pos "There is nothing left to prove.";
+
+  match tac.elt with
+  | P_tac_print
+  | P_tac_proofterm
+  | P_tac_query(_)      -> assert false (* Handled above. *)
+  | P_unif_solve        -> solve ps tac.pos
+  | _                   ->
+
+  (* Get the unif goals, the first type goal and the following goals *)
+  let pre_g,g,post_g =
```

Comment:
style: hanks to add a space after a comma.

Reply:
fixed in ed44999